plenoptic/simulate/canonical_computations/steerable_pyramid_freq.py

```diff
@@ -481,22 +501,38 @@ def convert_tensor_to_pyr(self, pyr_tensor):
         ----------
         pyr_tensor: `torch.Tensor` or `torch.ComplexTensor` (BxCxHxW)
             the pyramid coefficients
+        
+        pyr_info: List 
+            containing the number of channels, if split_complex was used
+            in the convert_pyr_to_tensor, and the list of pyramid keys for the dictionary
+            This should be taken from the output of convert_pyr_to_tensor
 
         Returns
         ----------
         pyr_coeffs: `OrderedDict`
             pyramid coefficients in dictionary format
         """
-
+        num_channels = pyr_info[0]
+        split_complex = pyr_info[1]
+        pyr_keys = pyr_info[2]
```

Comment:
I guess this info can't be inferred somehow? I don't love the idea of returning what's basically a dummy variable from one method to pass to this one, but if it can't be avoided...

I have a preference for splitting out these three values into three separate arguments, to make it clearer. But if the use case is always just grabbing the value returned from `convert_pyr_to_tensor` and passing it here, I guess it doesn't make too much of a difference. what do others think?

Reply:
well so the `pyr_keys` piece of the pyr_info list is a useful thing to return because it allows the user to matchup the keys with the tensor channel order. the other two things that are returned yes are more just for the paired `convert_tensor_to_pyr` function. They can't be inferred unless I turn these functions back into to class methods and change class attributes that stores these things when the convert function is called? Idk what you think is better..  And lastly yes I could split them- it just makes it easier to handle when you don't need them.